upgrade-order-processing.md

```diff
@@ -0,0 +1,357 @@
+These changes aim to enhance functionality, improve maintainability, and make implementing the changes easier.
+
+### What has changed?
+
+Introducing new middleware components to the order processing system instead of previously used order preview calculation.
+The main idea of a new processing system is to fill the `OrderData` object with all necessary data for the order creation, so the order creation itself is just a matter of use the `OrderData` to create the appropriate entity.
+
+The new order processing system is based on the middleware pattern.
+Each middleware is responsible for filling the `OrderData` object with a specific part of the order data.
+The order of the middleware is important, as each middleware can depend on the data filled by the previous middleware.
+
+You can configure the order of middlewares and which are used in the `app/config/packages/shopsys_framework.yaml` file.
+
+To upgrade fully to the new order processing system, you need to implement your own business logic into the middlewares and register them in the previously mentioned file.
+
+It's possible that your business logic is truly complex, and you may question the benefits of the new system.
+
+In that case, you may:
+
+-   Use the old order preview calculation (chances are you already have overloaded quite a portion of OrderPreview and the OrderPreviewCalculation classes).
+    You need to copy the old classes to your project and the order process will work as before.
+    But keep in mind that the OrderPreview system may introduce some problems during future upgrades.
+
+-   You may create a new middleware that will leverage the old order preview calculation system.
+    This way, you can keep the old system and still benefit from the new middleware system.
+    This may be a good way to gradually migrate to the new system.
+
+For more information about the order processing system, you may see the [Order Processing](https://docs.shopsys.com/introduction/order-processing/) article and the [Adding a New Order Item Type](https://docs.shopsys.com/cookbook/adding-a-new-order-item-type) cookbook.
+
+## BC Breaking changes
+
+-   The method `Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserFacade::__construct()` was changed:
+
+```diff
+ public function __construct(
+     // ...
+     protected readonly Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserRefreshTokenChainFacade $customerUserRefreshTokenChainFacade,
++    protected readonly Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressFactory $deliveryAddressFactory,
++    protected readonly Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressDataFactory $deliveryAddressDataFactory,
+ )
+```
+
+-   `Shopsys\FrameworkBundle\Model\Newsletter\NewsletterFacade`:
+    -   The method `addSubscribedEmail()` was removed.
+    -   The method `getNewsletterSubscriberById()` is now strictly typed.
+    -   The method `deleteById()` was removed.
```

Comment:
General question. Should it be mentioned that not only a method has been removed but also, if there is a replacement, what it is? Here, the example "the method 'deleteById()' was removed, use 'delete()' instead."?

Reply:
yup, that makes sense. Added to all removed bullets.